src/main/java/net/sf/jabref/logic/net/Cookie.java

```diff
@@ -123,12 +112,16 @@ public boolean matches(URI uri) {
             return false;
         }
 
-        String path = uri.getPath();
-        if (path == null) {
-            path = "/";
+        String uriPath = uri.getPath();
+        if (uriPath == null) {
+            uriPath = "/";
         }
 
-        return path.startsWith(this.path);
+        return uriPath.startsWith(this.path);
+    }
+
+    public boolean equals(Cookie cookie) {
+        return ((domain.equals(cookie.domain)) && (name.equals(cookie.name)));
```

Comment:
For unknown reasons (as in I didn't bother to dig deeper into it), it seems like equals is not overridden. It may be better to call it something different though. The main purpose was to make the fields private and the only place the fields were accessed was a when checking name and domain for equality.